crates/flowctl/src/raw/oauth.rs

```diff
@@ -0,0 +1,273 @@
+use crate::local_specs;
+use anyhow::{bail, Context};
+use itertools::Itertools;
+use proto_flow::{capture, flow};
+use std::{
+    collections::HashMap,
+    fmt::{Debug, Display},
+    net::ToSocketAddrs,
+    sync::{Arc, Mutex},
+};
+use warp::{http::Response, Filter};
+
+#[derive(Debug, clap::Args)]
+#[clap(rename_all = "kebab-case")]
+pub struct Oauth {
+    /// Path or URL to a Flow specification file.
+    #[clap(long)]
+    source: String,
+    /// Name of the task to test oauth for.
+    #[clap(long)]
+    task: Option<String>,
+    /// Docker network to run the connector, if one exists
+    #[clap(long, default_value = "bridge")]
+    network: String,
+
+    /// Your OAuth application's client ID. This corresponds to the
+    /// `connectors.oauth2_client_id` column in the database
+    #[clap(long)]
+    client_id: String,
+    /// Your OAuth application's client secret. This corresponds to the
+    /// `connectors.oauth2_client_secret` column in the database
+    #[clap(long)]
+    client_secret: String,
+    /// The endpoint config you want to test with. This is the config for a
+    /// sample user account that will perform the end-to-end consent flow
+    #[clap(long)]
+    endpoint_config: serde_json::Value,
+    /// If the connector's oauth spec required needs any other values injected.
+    /// This will match the `connectors.oauth2_injected_values` column in the database.
+    #[clap(long)]
+    injected_values: Option<serde_json::Value>,
+}
+
+pub async fn do_oauth(
+    ctx: &mut crate::CliContext,
+    Oauth {
+        source,
+        task,
+        network,
+        client_id,
+        client_secret,
+        endpoint_config,
+        injected_values,
+    }: &Oauth,
+) -> anyhow::Result<()> {
+    // TESTING
```

Comment:
The `tracing` stuff should already be setup in `main.rs`, so I don't think we'd need it here. Guessing this was intended to be removed?

Reply:
So I actually wonder what your opinion is here: currently it seems that we only really use `tracing` for debug logging, and we just do regular old `println!` for normal/informative output. That makes sense when we need to do things like print out tables or capture input, but I would love to be able to use e.g `tracing::info!` to log out informative messages. The problem is, we set up tracing to only log messages that are warn-level and higher, and it didn't seem straightforward to refactor that to be configurable on a per-command basis. Though as I write that, I'm doing exactly that here so I could just extract this `tracing_subscriber` override logic to a function that any command handler could call. 

Am I barking up the wrong tree by trying to use tracing here?

Reply:
The current state of affairs is pretty messy, and I don't think we follow any principle especially closely. So I'm open to other ideas, but here's how I've been thinking about it:

There's a difference between logging that we're putting in for debugging purposes, and that which is informational and meant to communicate to the user. IMO the main thing is that I want the code to differentiate those somehow, so that it's clear which one it is when you're reading it. For example, if we used `tracing` directly to output user-facing messages, then if you saw `tracing::info!("added specs to draft");`, it may not be clear whether that was meant as a user-facing message or not. IMO this isn't an especially strong argument, but 🤷 it's where we're at. So we've been just using `println!` for user-facing messages.

One possible improvement might be to define a `CliContext::info_message(&self, msg: &str)` function, which could delegate to either `println!` or `tracing::info!`. One other consideration is just that `tracing` outputs to stderr, while `println!` goes to stdout, but I'm not really sure how significant that is.

I think for this PR, we should just stick to the `println!` pattern, but I'd be open to trying to improve on this situation in a separate PR.